### PR TITLE
TW-1735: can't search matrix id

### PR DESCRIPTION
**Pull request by Te-Z**
Ticket: https://github.com/linagora/twake-on-matrix/issues/1735

Waiting for this PR to be merged: https://github.com/linagora/matrix-dart-sdk/pull/59

Demo:

[Capture vidéo du 20-05-2024 17:51:03.webm](https://github.com/linagora/twake-on-matrix/assets/31937920/24603954-5df5-42ef-b503-e3415bdee2f0)

### CODE REVIEW

**Review by hoangdat — APPROVED** [review]

**Review by nqhhdev — APPROVED** [review]

File: lib/pages/search/search_external_contact.dart [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  });
+
+  final String keyword;
+  final SearchController searchController;
+
+  @override
+  Widget build(BuildContext context) {
+    final newContact = PresentationContact(
+      matrixId: keyword,
+      displayName: keyword.substring(1),
```
Comment (sherlockvn) [inline lib/pages/search/search_external_contact.dart:23]: should check the keyword length first

**Comment by sherlockvn** [comment]
should we handle this case: @hoangdat 
image

**Comment by Te-Z** [comment]
Hi @hoangdat @sherlockvn , to handle keyword length and the case with `@@@@@@tom:client.com` I updated the isValidMatrixId method from sdk: https://github.com/linagora/matrix-dart-sdk/pull/59 
Those cases are handled and tested in unit test

File: lib/pages/search/search_external_contact.dart (hunk 1 above)
Comment (Te-Z) [reply lib/pages/search/search_external_contact.dart:23]: done in `isValidMatrixId`

**Review by sherlockvn — APPROVED** [review]

**Review by hoangdat — APPROVED** [review]

**Pull request merged**